gitlab/v4/cli.py

```diff
@@ -63,6 +67,17 @@ def __init__(
                     obj.set_from_cli(self.args[attr_name])
                     self.args[attr_name] = obj.get()
 
+    def _process_from_parent_attributes(self) -> None:
+        """Items in the path need to be url-encoded. There is a 1:1 mapping from
+        mgr_cls._from_parent_attrs <--> mgr_cls._path. Those values must be url-encoded
+        as they may contain a slash '/'."""
+        for key in self.mgr_cls._from_parent_attrs:
+            if key in self.args:
+                self.parent_args[key] = gitlab.utils.clean_str_id(self.args[key])
+                # If we don't delete it then it will be added to the URL as a
+                # query-string
+                del self.args[key]
```

Comment:
I'm probably a bit annoying with this but I like to apply it even for small things, then the comment fits :yum: then the comment fits

> [Flat is better than nested.](https://www.python.org/dev/peps/pep-0020/)

```suggestion
        for key in self.mgr_cls._from_parent_attrs:
            if not key in self.args:
                continue

            self.parent_args[key] = gitlab.utils.clean_str_id(self.args[key])
            # If we don't delete it then it will be added to the URL as a query-string
            del self.args[key]
```

Reply:
I like it. Thanks. Done.